### ISSUE TITLE
helm charts do not deploy in master on 1.7, 1.8, or 1.9 of Kubernetes

### ISSUE THREAD

**Issue by sdake** [istio/istio#2935]
```
sdake@falkor-07:~/go/src/istio.io/istio/install/kubernetes/helm$ helm install istio --name istio
Error: error unpacking initializer in istio: chart metadata (Chart.yaml) missing
```

**Comment by sdake** [comment]
Indeed the Chart.yaml file is missing from the initializer directory.

After placing a proper Chart.yaml in the initializer directory, a followup error is:
```
sdake@falkor-07:~/go/src/istio.io/istio/install/kubernetes/helm$ helm install --debug istio --name istio
[debug] Created tunnel using local port: '33769'

[debug] SERVER: "127.0.0.1:33769"

[debug] Original chart version: ""
[debug] CHART PATH: /home/sdake/go/src/istio.io/istio/install/kubernetes/helm/istio

Error: cannot load requirements: error unmarshaling JSON: json: cannot unmarshal array into Go value of type chartutil.Requirements
```

**Comment by sdake** [comment]
cc @linsun @luisyonaldo @lachie83 @sbezverk

**Comment by linsun** [comment]
I don't think we need the initializer dir, that dir should be removed.

**Comment by sdake** [comment]
I wasn't sure - I was OOO last week, so haven't kept up with the code base.

**Comment by luisyonaldo** [comment]
that was introduced by https://github.com/istio/istio/pull/2830

**Comment by luisyonaldo** [comment]
is not even a helm chart, is just a manifest definition inside the helm chart

**Comment by linsun** [comment]
With #2937, still failed with 

```
Error: render error in "istio/charts/sidecar-injector/templates/istio-sidecar-injector.yaml": template: istio/charts/sidecar-injector/templates/istio-sidecar-injector.yaml:2:14: executing "istio/charts/sidecar-injector/templates/istio-sidecar-injector.yaml" at <.Values.global.proxy...>: can't evaluate field debug in type interface {}``` 

will look into that next.

**Comment by sdake** [comment]
running `updateVersion.sh` (generating a new `values.yaml`) seems to move past the render error.

values.yaml does not properly seem to modify this tag.  I am unclear if this tag is necessary.
```
#
# initializer configuration
#
initializer:
  enabled: true
  serviceAccountName: default # used only if RBAC is not enabled
  replicaCount: 1
  imagePullPolicy: IfNotPresent
  resources: {}
  nodeSelector: {}

  initImage:
    repository: docker.io/istio/proxy_init
    tag: 0.4.0
```

**Comment by luisyonaldo** [comment]
https://github.com/istio/istio/pull/2965

**Comment by linsun** [comment]
leave this on for 1.9 testing cc @sdake but I've tested in 1.7 where istio did deploy without side car injector.

**Comment by sdake** [comment]
@linsun I missed your cc and opened a new issue to track the mutating webhook implementation in this issue: https://github.com/istio/istio/issues/3018

I guess  we can close this issue as resolved then?

**Comment by linsun** [comment]
SGTM

**Comment by morvencao** [comment]
Helm install `Istio-0.5.0` release still gets:
```
[llcao@mst helm]# helm install istio --name istio --namespace=istio-system
Error: error unpacking initializer in istio: chart metadata (Chart.yaml) missing
```
Would you help to update the release package? 
Or should I do helm install with “Canary” builds from the latest master branch for now?

**Comment by cfontes** [comment]
I am also having the same error on latest Minikube and HELM with latest ISTIO

**Comment by sdake** [comment]
@cfontes are you using Helm from Istio 0.5.0?  That is known to be non-functional.  0.5.1 may have enough of the fixes to get things going, however, if you have a desire to use the Helm charts, I'd recommend master for the time-being.

**Comment by sdake** [comment]
@morvencao right, that issue was fixed in master, but not fixed in time for 0.5.0, so you will see that error if using 0.5.0.  At this time the only thing I can recommend related to Helm is using master as several bugs and inconsistencies with the main manifests have been resolved (including the bug you see).

AFAIK helm works in master correctly.
